contracts/loan_manager/src/contract.rs

```diff
@@ -233,6 +240,36 @@ mod tests {
         Env,
     };
 
+    #[test]
+    fn deploy_pool() {
+        // ARRANGE
+        let e = Env::default();
+
+        // Setup test token
+        let admin = Address::generate(&e);
+        let token_address = e.register_stellar_asset_contract(admin.clone());
+        let ticker = Symbol::new(&e, "XLM");
+
+        let deployer_client = LoanManagerClient::new(&e, &e.register_contract(None, LoanManager));
+
+        let wasm_hash = e.deployer().upload_contract_wasm(loan_pool::WASM);
+        let salt = BytesN::from_array(&e, &[0; 32]);
+
+        // ACT
+        // Deploy contract using loan_manager as factory
+        let loan_pool_addr =
+            deployer_client.deploy_pool(&wasm_hash, &salt, &token_address, &ticker, &800_000);
+
+        // ASSERT
+        // No authorizations needed - the contract acts as a factory.
+        assert_eq!(e.auths(), &[]);
```

Comment:
Tää kommentti pöllitty esimerkeistä, mutten kyllä ymmärrä miksei tää tarvii authia? Mikä estää ketä tahansa kutsumasta deploy_pool?

https://developers.stellar.org/docs/build/smart-contracts/example-contracts/deployer

Reply:
Tossa esimerkissä sitä deployeria voi käyttää suoraan tai niin, että se deployaa toisen osoitteella. Silloin kun se deploy tehdään deployerilla niin toi auth ei varmaan siirry enviin kun se tapahtuu siinä sopimuksen sisällä. Ton linkin toisessa testissä taas e.auths() antaa ulkopuolisen deployerin authit.

Tällä hetkellähän meidän deploy_pool ei tosiaan tarvii mitään autentikointia. Me kuitenkin halutaan tulevaisuudessa lisätä tämäkin adminin taakse. Näitä pitää ihan ajan kanssa pohtia et mitä menee adminille, mitä voidaan tehdä vain ja ainoastaan kerran ja ne on sitten lukittuna ja mikä on julkisessa käytössä.